great_expectations/data_context/data_context.py

```diff
@@ -1141,20 +1066,60 @@ def store_validation_param(self, run_id, key, value):
         Returns:
             None
         """
-        self._evaluation_parameter_store.set(run_id, key, value)
+        run_params = self.get_parameters_in_evaluation_parameter_store_by_run_id(run_id)
```

Comment:
Not important for how we're doing this, but I find this confusing because the store is being required to store a dictionary, but is also being used as though it's a generic store, in the sense that we've got an enormous method name that is designed to do some amount of packing/unpacking.

Reply:
I agree.

I think the solution for this is to make Stores truly Namespace aware.

But to do that, I'd really want to implement the `ResourceIdentifier` classes that make up the broader namespace, and I was trying to not do that yet in the PR.

Reply:
👍